### PR TITLE
Fix to nu-scatter tallies

### PR DESCRIPTION
**Pull request by nelsonag**
As discussed in #460, the secondary bank changes introduced a bug where the nu-scatter tally scores were not tallying the scatter production rate, but only scattering rate.

The way it is implemented here is a bit of a kluge; I honestly am not sure if anyone uses the nu-scatter scores besides @wbinventor and myself (i.e., those using OpenMC for MGXS), so I chose to go with extra computations at tally time which only penalizes those using nu-scatter* scores as opposed to adding an attribute to the particle class which will penalize everyone.

### CODE REVIEW

**Comment by nelsonag** [comment]
Wow.... first time using the automated style checking; thats real cool.  Good job @smharper!

File: src/tally.F90 [hunk 1]
```diff
@@ -170,10 +172,32 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
...
+            if (p % event_MT == nuclides(p % event_nuclide) % reactions(m) % MT) then
+              ! Found the reaction, set our pointer and move on with life
+              rxn => nuclides(p % event_nuclide) % reactions(m)
+              exit
+            end if
+          end do
+
+          ! Get multiplicity
+          if (rxn % multiplicity_with_E) then
+            multiplicity = interpolate_tab1(rxn % multiplicity_E, p % last_E)
```
Comment (smharper) [inline src/tally.F90]: If `rxn % multiplicity_with_E` then we increased the particle weight.  We can skip the table interpolation and just use `score = p % wgt`

**Comment by smharper** [comment]
:D I'm glad you like the style checker and don't find it too authoritarian.

**Comment by smharper** [comment]
Good job uncovering this bug @nelsonag and @wbinventor!  This could easily have gone unnoticed for a long time.

**Comment by nelsonag** [comment]
@smharper good catch.  Fixed it as you suggested.  Note the discussio nat #460, possibly not ready to merge until we identify whats going on with scatter scores as well.

File: src/tally.F90 [hunk 2]
```diff
@@ -170,10 +171,34 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
...
+          end do
+
+          ! Get multiplicity and apply to score
+          if (rxn % multiplicity_with_E) then
+            ! Then the multiplicity was already incorporated in to p % wgt
+            ! per the scattering routine,
+            score = p % wgt
+          else
+            ! Grab the multiplicity from the rxn
+            score = p % last_wgt * real(rxn % multiplicity,8)
```
Comment (paulromano) [inline src/tally.F90]: `rxn % multiplicity` will get promoted to real(8) anyway, so no need to explicitly cast it.

File: src/tally.F90 [hunk 3]
```diff
@@ -183,10 +208,34 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
...
```
Comment (paulromano) [inline src/tally.F90]: Same thing here

File: src/tally.F90 [hunk 4]
```diff
@@ -196,10 +245,34 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
...
```
Comment (paulromano) [inline src/tally.F90]: and here

File: src/tally.F90 [hunk 5]
```diff
@@ -5,6 +5,7 @@ module tally
   use error,            only: fatal_error
   use geometry_header
   use global
+  use interpolation
```
Comment (paulromano) [inline src/tally.F90]: Don't think this is needed now

**Comment by paulromano** [comment]
@nelsonag Thanks for submitting this fix. After thinking about this whole issue for a few hours, I'm ok with the fix you have here and I don't believe any other scores are incorrect as I initially thought. The one thing that I will request is that you change the test_score_nuscatter* tests so that they don't have any inactive batches (and then update the results). Since your PR changes the nu-scatter score, one would hope the nu-scatter tests pick it up, but as is right now (n,2n) only happens in inactive batches so the results didn't actually change.

**Comment by nelsonag** [comment]
Thanks for the review @paulromano! Changes factored in, pending completion of travis-ci it seems like this should be ready to go.

**Comment by paulromano** [comment]
And thank you @nelsonag for an important fix!

**Pull request merged**